File: RETOS/Reto02/Reto02.java
package RETOS.Reto02;
import java.util.Scanner;
public class Reto02 {
/*
 * Escribe un programa que reciba un solo carácter por parte del usuario y muestre por consola (con un print) si dicho carácter es una vocal o una consonante. Si el carácter no es una letra, el programa deberá mostrar un mensaje de error.
 *
 * - Si el carácter es una vocal, mostrará: "Es una vocal".
 * - Si el carácter es una consonante, mostrará: "Es una consonante".
 * - Si el carácter no es una letra (números, símbolos, espacios, etc.), mostrará: "Error: No es una letra".
 *
 *El programa debe funcionar tanto para letras mayúsculas como minúsculas.
 */
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingresa un solo carácter: ");
        String input = scanner.nextLine();

        // Verificar que se ingresó un solo carácter
        if (input.length() != 1) {
            System.out.println("Error: Se debe ingresar solo un carácter.");
            return;
        }

        char caracter = input.charAt(0);

        // Convertir a minúscula para simplificar la comparación
        char c = Character.toLowerCase(caracter);

        // Verificar si es una letra
        if (Character.isLetter(c)) {
            if ("aeiou".indexOf(c) >= 0) {
                System.out.println("Es una vocal.");
            } else {
                System.out.println("Es una consonante.");
            }
        } else {
            System.out.println("Error: No es una letra.");
        }
    }
}

File: RETOS/Reto2/Reto02.js
/*
*Escribe un programa que reciba un solo carácter por parte del usuario y muestre por consola (con un print) si dicho carácter es una vocal o una consonante. Si el carácter no es una letra, el programa deberá mostrar un mensaje de error.

* - Si el carácter es una vocal, mostrará: "Es una vocal".
* - Si el carácter es una consonante, mostrará: "Es una consonante".
* - Si el carácter no es una letra (números, símbolos, espacios, etc.), mostrará: "Error: No es una letra".

*El programa debe funcionar tanto para letras mayúsculas como minúsculas.
*/
const readline = require('readline');

// Crear una interfaz de lectura para capturar entrada del usuario
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const clasificarCaracter = (caracter) => {
    // Convertir a minúscula para simplificar la comparación
    const c = caracter.toLowerCase();

    // Verificar si es una letra
    if (/[a-z]/.test(c)) {
        if ('aeiou'.includes(c)) {
            return "Es una vocal.";
        } else {
            return "Es una consonante.";
        }
    } else {
        return "Error: No es una letra.";
    }
};

// Pedir la entrada del usuario
rl.question('Ingresa un solo carácter: ', (input) => {
    if (input.length !== 1) {
        console.log("Error: Se debe ingresar solo un carácter.");
    } else {
        console.log(clasificarCaracter(input));
    }
    rl.close(); // Cerrar la interfaz de lectura
});
